src/pz390.java

```diff
@@ -8129,12 +8130,34 @@ private void ins_B9XX(){
 				reg.putLong(rf1, rlv3);
 				psw_cc = get_long_log_sub_cc();
 				break;
-	     case 0xEC: // "B9EC=MGRK,54,340", // B9EC rrfa MGRK R1,R2,R3 RPI 2202
+	     case 0xEC: // "B9EC=MGRK,54,340", // B9EC rrfa MGRK R1,R2,R3 RPI 2202 #300
+		     psw_check = false;
 	    	 ins_setup_rrfa();
+			 if ((mf1 & 1) != 0){ // issue 300 add MGRK, MSGRKC, MSRKC
+				set_psw_check(psw_pic_spec);
+			}
+			big_int1 = new BigInteger(get_signed_bytes(reg_byte, rf2, 8)); 
+																			
+			big_int2 = new BigInteger(get_signed_bytes(reg_byte, rf3, 8));
+			big_int1 = big_int1
+			         .multiply(big_int2);
+			fp_bi_to_wreg(reg_byte,rf1,big_int1, 16); 
 	    	 break;
-	    case 0xED: // "B9ED=MSGRKC,54,340", // B9ED rrfa MSGRKC R1,R2,R3 RPI 2202
+	    case 0xED: // "B9ED=MSGRKC,54,340", // B9ED rrfa MSGRKC R1,R2,R3 RPI 2202 #300
+			 psw_check =false;
 	    	 ins_setup_rrfa();
-	    	 break;
+			 big_int1 = new BigInteger(get_signed_bytes(reg_byte, rf2, 8)); 
+																			
+			big_int2 = new BigInteger(get_signed_bytes(reg_byte, rf3, 8));
+			big_int1 = big_int1
+			         .multiply(big_int2);
+					 if (big_int1.compareTo(bi_max_pos_long) == 1
+					     || big_int1.compareTo(bi_min_neg_long) == -1) {                  
+				psw_cc = psw_cc3;                          
+				break;                                     
+			} 
+			fp_bi_to_wreg(reg_byte,rf1,big_int1, 8);
+	    	break;
```

Comment:
I thought I added comments here a few days ago. They're gone??? Repeat them now.

MSGRKC sets CC 0,1,2,3. Only CC 3 set in above code.

MSGRKC (and MSC, MSGC, MSRKC) can result in fixed-point overflow exception if PSW corresponding Program Mask bit is set. See code for AR instruction on how CC is set, accounting for this possibility.